testing/tests/test_e2e/test_state.py

```diff
@@ -984,11 +998,11 @@ class Charm(ops.CharmBase):
         relations=iterable({relation}),
         stored_states=iterable({stored_state}),
     )
-    assert len(state.containers) == 2
     assert isinstance(state.containers, frozenset)
-    assert len(state.relations) == 5
+    assert len(state.containers) == 2
     assert isinstance(state.relations, frozenset)
-    assert len(state.storages) == 1
+    assert len(state.relations) == 5
     assert isinstance(state.storages, frozenset)
-    assert len(state.stored_states) == 1
+    assert len(state.storages) == 1
     assert isinstance(state.stored_states, frozenset)
+    assert len(state.stored_states) == 1
```

Comment:
This looks odd in the diff but what it's doing is moving the `isinstance` assertion above the `len` one, because `len(Iterable)` can't be done, so we need to verify that it has a `__len__` first.